lib/ts/recipe/dashboard/api/multitenancy/createOrUpdateThirdPartyConfig.ts

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { APIInterface, APIOptions } from "../../types";
+import Multitenancy from "../../../multitenancy";
+import SuperTokensError from "../../../../error";
+
+export type Response =
+    | {
+          status: "OK";
+          createdNew: boolean;
+      }
+    | {
+          status: "INVALID_PROVIDER_CONFIG";
+      };
+
+export default async function createOrUpdateThirdPartyConfig(
+    _: APIInterface,
+    __: string,
+    options: APIOptions,
+    userContext: any
+): Promise<Response> {
+    const requestBody = await options.req.getJSONBody();
+    const { tenantId, providerConfig } = requestBody;
+
+    if (typeof tenantId !== "string" || tenantId === "") {
+        throw new SuperTokensError({
+            message: "Missing required parameter 'tenantId'",
+            type: SuperTokensError.BAD_INPUT_ERROR,
+        });
+    }
+
+    let thirdPartyRes;
+
+    try {
+        thirdPartyRes = await Multitenancy.createOrUpdateThirdPartyConfig(
+            tenantId,
+            providerConfig,
+            undefined,
+            userContext
+        );
+    } catch (_) {
+        return {
```

Comment:
Does this throw for invalid config? Would it be possible to include the error message? I think core reports why the config is invalid